components/resource_control/src/resource_group.rs

```diff
@@ -267,15 +281,19 @@ impl ResourceController {
                 // but it should be ok as this operation should be extremely rare
                 // and the impact is not big.
                 if near_overflow {
-                    tracker.decrease_vt(RESET_VT_THRESHOLD - (max_vt - vt) / 2);
+                    tracker.decrease_vt(RESET_VT_THRESHOLD);
```

Comment:
I think the essential problem is  the growth rate of `vt` be very fast, especially for relatively `hot group`, then the `update_min_virtual_time` cannot catch up it to avoid the overflow rate. maybe we can try to decrease the weight factor with some normalization methods?

Reply:
The maximum weight is MaxInt32 when the ru_per_sec = 1, in this case if a resource_group is overflow in 1s, the resource consumation should be 2^60/2/2^31 = 2^28 that is about 268M. This is not a very big number as our cpu time is managed in microseconds, so in the extreme case, if a resource group consumes more than 268 seconds cpu or write more than 268M bytes within 1s, the `update_min_virtual_time` is not fast enough to handle.  
In my test, with ru_per_sec = 1 and 8c tikv server, it takes a read workload about 1min to trigger this overflow.

BTW, the PR here is to fix the bug that because the `vt` is load after `max_vt`, in my test with 128 groups, the duration is about 10us, so it is possible some resource group's `vt` has grown bigger than the `max_vt`, thus `max-vt` is negatively overflow and  cause `RESET_VT_THRESHOLD - (max_vt - vt) / 2` to be about `u64::MAX/2`.

I original thought this overflow is not triggered very often because it has some side effect that can impact the task lantency because of task priority's disorder. Maybe we can rollback to the old algorithm to use a dynamic task wight formula and special handle the default weight if it is the default maximum value.